doc/index.md

```diff
@@ -331,6 +331,157 @@ to inject things into the public interface of the iterator:
     };
 ~~~~~~~
 
+From within mixin you can call:
+* `get()` - to access adaptor internals 
+* `base()` - to access adopted iterator
+
+Iterator/sentinel adaptor may "override" following members:
+~~~~~~~{.cpp}
+    class adaptor : public ranges::adaptor_base
+    {
+        // !For begin_adaptor only!
+        template<typename Rng>
+        constexpr decltype(auto) begin(Rng &rng)
```

Comment:
s/`decltype(auto)`/`auto`/

Reply:
??

Reply:
Yes, really.

Reply:
`??` = don't understand you completely.

----

`s/decltype(auto)/auto/` how this translates to English?

Reply:
Oh! I'm so sorry. I was speaking sed. /facepalm. Let me try English. Substitute ("s/") `auto` for `decltype(auto)`. In other words:

```suggestion
        constexpr auto begin(Rng &rng)
```